tools/protodoc/protodoc.py

```diff
@@ -177,22 +182,19 @@ def github_url(text, type_context):
     return f":repo:`{text} <api/{type_context.source_code_info.name}#L{type_context.location.span[0]}>`"
 
 
-def format_comment_with_annotations(comment, type_name=''):
+def format_comment_with_annotations(comment, show_wip_warning=False):
     """Format a comment string with additional RST for annotations.
 
     Args:
         comment: comment string.
-        type_name: optional, 'message' or 'enum' may be specified for additional
-           message/enum specific annotations.
+        show_wip_warning: whether to show the work in progress warning.
 
     Returns:
         A string with additional RST from annotations.
     """
-    alpha_warning = ''
-    if annotations.ALPHA_ANNOTATION in comment.annotations:
-        experimental_warning = (
-            '.. warning::\n   This API is alpha and is not covered by the :ref:`threat model <arch_overview_threat_model>`.\n\n'
-        )
+    wip_warning = ''
+    if show_wip_warning:
+        wip_warning = WIP_WARNING
```

Comment:
I think the original (which I probably added) for alpha is actually wrong. The API status is not the same as the extension implementation status.

Reply:
Sorry what do you mean? This I think is just related to the API alpha/wip conversion?

Reply:
The WiP warning is:
```
This API is work in progress and is not covered by the :ref:`threat model <arch_overview_threat_model>` 
```

What I'm saying is that even before, with alpha, the confusion of alpha for API and alpha for extension status is confusing. I think these notifications should be driven by extension maturity status not by API. Alpha API probably implies alpha implementation status but not the other way around, so this is under approximating.

Reply:
Sorry for being dense but I still don't understand what change you are asking for. I believe this is only triggering for the wip annotation I added? I think extension is handled elsewhere?

Reply:
I'm suggesting it's probably not the right place to be putting in this warning at all.

Reply:
This function is called from all the places I need it so it seems fine to me? Can you tell me specifically what you are suggesting I change?

Reply:
Sorry if I'm not being very clear. Here is my position:
* APIs do not have threat model coverage, extensions do.
* Since we are generating these warning based on API, we are missing warning about extensions that have stable APIs but alpha status in their extension implementation status (this information is in a different YAML file / location, CC @phlax). I think we have other warnings that derive from that metadata.

Maybe the wording is just confusing here. If instead we said something like `This API includes features that are currently work-in-progress. Those features marked as work-in-progress in the API are not covered by the Envoy threat model` then we have some value in these additional warning.

Reply:
But this applies to core also? I'm really confused how extensions are getting conflated here. There are other warnings for alpha extensions.

I can update the text to make it clear that it's not API wide, but I don't think extension should be included and we also need to include verbiage about breaking changes per the unified text I am now using.

I will update the unified text.

Reply:
I see, good point. Yeah, if you can just make clear it's not API-wide and that it's really a case of implementation status not being finalized until the API is that would be great.